ios/RCTWebRTC/WebRTCModule+VideoTrackAdapter.m

```diff
@@ -0,0 +1,178 @@
+
+#import <Foundation/Foundation.h>
+#include <libkern/OSAtomic.h>
+#import <objc/runtime.h>
+
+#import <React/RCTBridge.h>
+#import <React/RCTEventDispatcher.h>
+
+#import <WebRTC/RTCVideoRenderer.h>
+#import <WebRTC/RTCVideoTrack.h>
+
+#import "WebRTCModule.h"
+#import "WebRTCModule+RTCPeerConnection.h"
+#import "WebRTCModule+VideoTrackAdapter.h"
+
+/* Mute detection timer intervals. The initial timeout will be longer to
+ * accommodate for source startup.
+ */
+static const NSTimeInterval INITIAL_MUTE_DELAY = 3;
+static const NSTimeInterval MUTE_DELAY = 1.5;
+
+/* Entity responsible for detecting track mute / unmute events. It's implemented
+ * as a video renderer, which counts the number of frames, and if it sees them
+ * stalled for the default interval it will emit a mute event. If frame keep
+ * being received, the track unmute event will be emitted.
+ */
+@interface TrackMuteDetector : NSObject<RTCVideoRenderer>
+
+@property (copy, nonatomic) NSNumber *peerConnectionId;
+@property (copy, nonatomic) NSString *streamReactTag;
+@property (copy, nonatomic) NSString *trackId;
+@property (weak, nonatomic) WebRTCModule *module;
+
+@end
+
+@implementation TrackMuteDetector {
+    BOOL _disposed;
+    volatile int64_t _frameCount;
+    BOOL _muted;
+    dispatch_source_t _timer;
+}
+
+- (instancetype)initWith:(NSNumber*)peerConnectionId
+          streamReactTag:(NSString*)streamReactTag
+                 trackId:(NSString*)trackId
+            webRTCModule:(WebRTCModule*)module {
+    self = [super init];
+    if (self) {
+        self.peerConnectionId = peerConnectionId;
+        self.streamReactTag = streamReactTag;
+        self.trackId = trackId;
+        self.module = module;
+
+        _disposed = NO;
+        _frameCount = 0;
+        _muted = NO;
+        _timer = nil;
+    }
+
+    return self;
+}
+
+- (void)dispose {
+    _disposed = YES;
+    if (_timer != nil) {
+        dispatch_source_cancel(_timer);
+        _timer = nil;
+    }
+}
+
+- (void)emitMuteEvent:(BOOL)muted {
+    [self.module.bridge.eventDispatcher
+        sendDeviceEventWithName:@"peerConnectionSignalingStateChanged"
```

Comment:
This event name is incorrect, so the whole thing does not work, but I'll fix it since I'm at it.

Reply:
it appears that just changing the name will not fix the issue, see other comment below about missing part

Reply:
what comments below? I can't see any about missing stuff.